src/Fixer/LanguageConstruct/NullableTypeDeclarationFixer.php

```diff
@@ -0,0 +1,335 @@
+    public null|bool $internal;
+    public null|\Closure $callback;
+}
+',
+                    new VersionSpecification(8_00_00),
+                    ['syntax' => self::OPTION_SYNTAX_QUESTION_MARK]
+                ),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return \PHP_VERSION_ID >= 8_00_00 && $tokens->isTokenKindFound($this->candidateTokenKind);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before OrderedTypesFixer, TypesSpacesFixer.
+     * Must run after NullableTypeDeclarationForDefaultNullValueFixer, SingleSpaceAroundConstructFixer.
+     */
+    public function getPriority(): int
+    {
+        return 2;
+    }
+
+    public function configure(array $configuration): void
+    {
+        parent::configure($configuration);
+
+        $this->candidateTokenKind = self::OPTION_SYNTAX_QUESTION_MARK === $this->configuration['syntax']
+            ? CT::T_TYPE_ALTERNATION // `|` -> `?`
+            : CT::T_NULLABLE_TYPE; // `?` -> `|`
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('syntax', 'Whether to use question mark (`?`) or explicit `null` union for nullable type.'))
+                ->setAllowedValues([self::OPTION_SYNTAX_UNION, self::OPTION_SYNTAX_QUESTION_MARK])
+                ->setDefault(self::OPTION_SYNTAX_QUESTION_MARK)
+                ->getOption(),
+        ]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        $functionsAnalyzer = new FunctionsAnalyzer();
+
+        foreach (array_reverse($this->getElements($tokens), true) as $index => $type) {
+            if ('property' === $type) {
+                $this->normalizePropertyType($tokens, $index);
+
+                continue;
+            }
+
+            // fix return before arguments
+            $this->normalizeMethodReturnType($functionsAnalyzer, $tokens, $index);
+            $this->normalizeMethodArgumentType($functionsAnalyzer, $tokens, $index);
+        }
+    }
+
+    /**
+     * @return array<int, string>
+     *
+     * @phpstan-return array<int, 'method'|'property'>
+     */
+    private function getElements(Tokens $tokens): array
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        $elements = array_map(
+            static fn (array $element): string => $element['type'],
+            array_filter(
+                $tokensAnalyzer->getClassyElements(),
+                static fn (array $element): bool => \in_array($element['type'], ['method', 'property'], true)
+            )
+        );
+
+        foreach ($tokens as $index => $token) {
+            if (
+                $token->isGivenKind(T_FN)
+                || ($token->isGivenKind(T_FUNCTION) && !isset($elements[$index]))
+            ) {
+                $elements[$index] = 'method';
```

Comment:
```suggestion
                $elements[$index] = 'function';
```

Every method is a function, while not every function is a method 🙂. If we include arrow functions here, we should stick with `function` because arrow function can't be a method. If we need single type for all functions, this would also require change in `array_map()` above and `@phpstan-return` type declaration. Otherwise only `@phpstan-return` should be extended with third possible value.